test/spmd/test_xla_distributed_checkpoint.py

```diff
@@ -208,6 +208,25 @@ def test_state_dict_separation(self):
           planner.unsharded_state_dict,
           ['fc1.weight', 'fc1.bias', 'fc2.weight', 'fc2.bias'])
 
+  def test_save_state_dict_with_cpu_shards(self):
+    model = self._get_sharded_model()
+    planner = self._get_save_planner(model)
```

Comment:
We want to create the SavePlanner with the result of `_sharded_cpu_state_dict` instead of the model's state_dict, so we shouldn't use this helper function. Something like this would work:

```
    sharded_state_dict = _sharded_cpu_state_dict(model.state_dict())
    planner = SPMDSavePlanner()
    planner.set_up_planner(sharded_state_dict, True)
````

Also, let's remove the `_is_sharded_tensor` check on line 214, since if n_devices > 1, fc1.weight should be _CpuShards in the state_dict.

Reply:
this is what I had earlier but the tests failed so I changed it